pkg/segment/results/blockresults/blockresult.go

```diff
@@ -469,6 +470,35 @@ func (gb *GroupByBuckets) ConvertToAggregationResult(req *structs.GroupByRequest
 				valIdx := gb.reverseMeasureIndex[idx]
 				finalVal := bucket.runningStats[valIdx].hll.Estimate()
 				currRes[mInfoStr] = utils.CValueEnclosure{CVal: finalVal, Dtype: utils.SS_DT_UNSIGNED_NUM}
+			case utils.Values:
+				valIdx := gb.reverseMeasureIndex[idx]
+				rawValStrArr, ok := bucket.runningStats[valIdx].rawVal.CVal.([]string)
+				if !ok {
+					currRes[mInfoStr] = utils.CValueEnclosure{CVal: nil, Dtype: utils.SS_INVALID}
+					continue
+				}
+
+				uniqueSet := make(map[string]struct{})
+				uniqueStrings := make([]string, 0)
+
+				for _, str := range rawValStrArr {
+					if _, exists := uniqueSet[str]; !exists {
+						uniqueSet[str] = struct{}{}
+						uniqueStrings = append(uniqueStrings, str)
+					}
+				}
+
+				sort.Strings(uniqueStrings)
+
+				strVal := ""
+				for _, str := range uniqueStrings {
+					strVal += (str + "&nbsp")
+				}
```

Comment:
nit: Can you make it so there's no `"&nbsp"` after the last element? You should be able to use `strings.Join(...)` for this. Then you'll also need to update ingest.csv and restart.csv for your new tests to work